bin/ch/ch.cpp

```diff
@@ -520,6 +514,12 @@ HRESULT RunScript(const char* fileName, LPCSTR fileContents, size_t fileLength,
                 IfFailGo(messageQueue->ProcessAll(fileName));
             } while(!messageQueue->IsEmpty());
         }
+
+        // free the source for the serialized script case if it's not been handed to a managed object
+        if (!serializedCallbackInfo.freeingHandled && fileContentsFinalizeCallback != nullptr)
+        {
+            fileContentsFinalizeCallback((void*)fileContents);
```

Comment:
Freeing serialised script source had to be moved down below the PrintException call - as GetExceptionWithMetaData accesses the source.